packages/host/app/services/realm-server.ts

```diff
@@ -87,6 +91,69 @@ export default class RealmServerService extends Service {
     window.localStorage.removeItem(sessionLocalStorageKey);
   }
 
+  get availableRealmURLs() {
+    return [...this._userRealmURLs, ...this._catalogRealmURLs];
+  }
+
+  get userRealmURLs() {
+    return this._userRealmURLs.filter((realmURL) => realmURL != baseRealm.url);
+  }
+
+  get catalogRealmURLs() {
+    return this._catalogRealmURLs;
+  }
+
+  async setAvailableRealmURLs(userRealmURLs: string[]) {
+    userRealmURLs.forEach((userRealmURL) => {
+      if (!this._userRealmURLs.includes(userRealmURL)) {
+        this._userRealmURLs.push(userRealmURL);
+      }
+    });
+
+    this._userRealmURLs.forEach((userRealmURL) => {
+      if (!userRealmURLs.includes(userRealmURL)) {
+        this._userRealmURLs.splice(
+          this._userRealmURLs.indexOf(userRealmURL),
+          1,
+        );
+      }
+    });
+
+    let baseRealmUrl = baseRealm.url;
+
+    if (!this._userRealmURLs.includes(baseRealmUrl)) {
+      this._userRealmURLs.unshift(baseRealmUrl);
+    }
+
+    await this.fetchCatalogRealmURLs();
+  }
+
+  private async fetchCatalogRealmURLs() {
+    let realmServerURL = new URL(window.location.href);
+    if (
+      realmServerURL.origin === 'http://localhost:4200' ||
+      realmServerURL.origin === 'http://localhost:7357'
```

Comment:
Where is 7357 coming from?

Reply:
i believe 7357 is the port that ember CLI runs on in CI. But we should really be using this.url here